src/main/kotlin/no/fdk/resourceservice/service/UnionGraphService.kt

```diff
@@ -356,6 +387,92 @@ class UnionGraphService(
         }
     }
 
+    /**
+     * Extracts DataService URIs from a dataset JSON-LD and adds their graphs to the union model.
+     *
+     * Follows the path: distribution[*].accessService[*].uri
+     * Only adds DataServices that haven't been expanded yet to avoid duplicates.
+     *
+     * @param datasetJsonLd The dataset JSON-LD map
+     * @param unionModel The union model to add DataService graphs to
+     * @param expandedUris Set of URIs that have already been expanded (modified in place)
+     */
+    private fun extractAndAddDataServices(
+        datasetJsonLd: Map<String, Any>,
+        unionModel: org.apache.jena.rdf.model.Model,
+        expandedUris: MutableSet<String>,
+    ) {
+        try {
+            // Extract distributions from the dataset
+            val distributions =
+                when (val distValue = datasetJsonLd["distribution"]) {
```

Comment:
Dette vil ikke være mulig via jsonLd, må bruke den interne datamodellen for at dette skal fungere